### PR TITLE
[feat] Add Notes tab to single run page

### PR DESCRIPTION
**Pull request by arsengit**
- Added `Notes Tab`
- Added `Notes Tab` link in run overview tab sidebar
- Added `Toast UI` WYSIWYG editor
- Added `NotesNotificationsEnum` enum
- Added `IApiRequestRef<T> ` interface for request references
- Deleted unnecessary rendered paragraph in `ConfirmModal` component

### CODE REVIEW

**Comment by arsengit** [comment]
Resolves #291

File: aim/web/ui/src/pages/RunDetail/NotesTab/NotesTab.scss [hunk 1]
```diff
@@ -0,0 +1,84 @@
+@use 'styles/abstracts/index' as *;
+
+.NotesTab {
+  .isLoading {
+    .toastui-editor-main {
+      opacity: 0;
+    }
+  }
+  &__Editor {
```
[the remark this answers is not in the document]
Comment (arsengit) [reply aim/web/ui/src/pages/RunDetail/NotesTab/NotesTab.scss:9]: @rubenaprikyan  Did we have a discussion about that?

[the remark this answers is not in the document]
Comment (rubenaprikyan) [reply aim/web/ui/src/pages/RunDetail/NotesTab/NotesTab.scss:9]: I am not sure we have documented decision summary, but had many small discussions, and using/not using BEM is a well known problem for us, and maybe this was the right time to ask for this , , ,the discussion/decision will be ASAP.

**Review by Hamik25 — APPROVED** [review]

**Review by rubenaprikyan — APPROVED** [review]
lgtm

**Review by roubkar — APPROVED** [review]

**Pull request merged**